### ISSUE TITLE
[BUG]: Burnout 3: Regression in rendering in latest nightly 

### ISSUE THREAD

**Issue by jack23247** [PCSX2/pcsx2#11482]
### Describe the Bug

> **Note**
> The problem disappears if I try rendering at native resolution; I decided to report it anyway since it is a regression compared to the behavior of the previous nightlies. Feel free to close this issue if you believe it is a duplicate and/or this is an already known problem.

Since nightly v1.7.5941, heavy vignetting is present in Burnout 3 with Automatic Hardware Renderer Fixes enabled on the DX11 and DX12 renderers (Vulkan and OpenGL seem to be unaffected); I believe the problem also introduces slight inconsistencies in frame times. I tested the following versions as I had them on hand: SLES-52585 (PAL), SLUS-21050 (NTSC-U Beta), and DNAT-20045 (Pre-Alpha), and they all show this issue.

The problem seems to be related to _Upscaling Fixes -> Native Scaling_, as it presents when this option is set to either "Normal" or "Aggressive". Setting this option to "Off" removes the vignetting but reintroduces another visual bug (duplicated bloom effects) which was solved by this rendering fix.

### Reproduction Steps

Play Burnout 3 in nightly v1.7.5941 with either the DX11 or DX12 renderer, default options and non-native resolution.

### Expected Behavior

No vignetting.

### PCSX2 Revision

v1.7.5941

### Operating System

Windows 10 (64bit)

### CPU

Intel Core i7-3770K

### GPU

AMD Radeon RX570

### GS Settings

![image](https://github.com/PCSX2/pcsx2/assets/35559767/b05eb3e3-facc-49f6-904c-708fac65e9f9)
![image](https://github.com/PCSX2/pcsx2/assets/35559767/3d5e67a9-4f83-48c3-8bdf-f86ea161805d)
![image](https://github.com/PCSX2/pcsx2/assets/35559767/5c9f9bad-8bbe-489b-bc7d-cc399ff47caf)

Whatever's not pictured has been left at its default value (besides show FPS and Frame Times under OSD settings). The global settings have been reset to their default values as well.

### GS Window Screenshots

![image](https://github.com/PCSX2/pcsx2/assets/35559767/9eaad0fe-a1c3-4f5c-b67d-f880fa91121e)
Native Scaling: Aggressive

![image](https://github.com/PCSX2/pcsx2/assets/35559767/51fb8565-d88e-476c-9440-5c01f2f92e5d)
Native Scaling: Off

### Logs & Dumps

[Burnout 3 - Takedown_SLES-52585_20240628153611_Native_Scaling_Bug.zip](https://github.com/user-attachments/files/16030432/Burnout.3.-.Takedown_SLES-52585_20240628153611_Native_Scaling_Bug.zip)

**Comment by refractionpcsx2** [comment]
can you correct the version number please. 1.7.594 was like 3 years ago.

**Comment by jack23247** [comment]
Sorry, I dropped a digit. 

I'd also like to add that my description of what happens when Native Scaling is Off ("duplicated bloom effects") is imprecise at best. I can try giving a better explanation (along with some pictures) if needed.

**Comment by refractionpcsx2** [comment]
so 5940 was fine?

**Comment by jack23247** [comment]
I updated PCSX2 earlier today, but I may have skipped a revision or two... is there a way to see which build I was running previously? I looked in `updater.log`, but there doesn't seem to be a version number listed. 

The UI says the last time I played, bar today, was 25/06, meaning I was running either .5396 or .5397... I'm not sure I played B3 on that occasion, though. I can try reinstalling .5396 and testing the game on each build to see where it broke.
